### PR TITLE
ARROW-15924: [Ruby] Add support for #values of MonthDayNanoInterval type

### PR DESCRIPTION
**Pull request by okadakk**

### CODE REVIEW

File: ruby/red-arrow/ext/arrow/converters.hpp [hunk 1]
```diff
@@ -320,6 +336,7 @@ namespace red_arrow {
     // VISIT(Interval)
```
Comment (kou) [inline ruby/red-arrow/ext/arrow/converters.hpp:336]: Can we remove this comment?

Comment (okadakk) [reply ruby/red-arrow/ext/arrow/converters.hpp]: Ok! I removed it!

File: ruby/red-arrow/ext/arrow/converters.hpp [hunk 2]
```diff
@@ -761,8 +773,6 @@ namespace red_arrow {
     VISIT(Time32)
     VISIT(Time64)
     VISIT(Timestamp)
-    // TODO
-    // VISIT(Interval)
```
Comment (kou) [inline ruby/red-arrow/ext/arrow/converters.hpp:765]: Ah, we need to implement dictionary array cases with tests.
Could you revert this change and open a new JIRA issue for adding tests of `Arrow::DictionaryArray#values`

Comment (okadakk) [reply ruby/red-arrow/ext/arrow/converters.hpp:765]: @kou 
sure!
I opened two JIRA issue!

https://issues.apache.org/jira/browse/ARROW-15935
https://issues.apache.org/jira/browse/ARROW-15936

**Review by kou — APPROVED** [review]
+1
